### DISCUSSION TITLE
How to remove a stamp and marks from an image?

### DISCUSSION THREAD

**Discussion by mediotex** [ImageMagick/ImageMagick#6522] [Help] [answered]
### ImageMagick version

6.9.11-60

### Operating system, version and so on

Debian 11

### Description

The scanned document image (grayscale) have two handwritten marks on the top and seal stamp at the bottom. I want to remove these marks and seal by replace them (hide) with same size rectangles, copied from another region of the image (part of background). How to do this with Imagemagick? 

![image](https://github.com/ImageMagick/ImageMagick/assets/93042540/7b01e8ac-fb34-4ac5-9d0d-a4dbecabffe5)

**Comment by fmw42** [comment]
Please post a real PDF so we can see the text and seal.  Are they always in the same area (rectangle dimensions the same and placement the same)?  Is the PDF density always the same?

**Comment by mediotex** [comment]
Only one, title page. Gray background subsequently should be removed, for printing.
![My project-1](https://github.com/ImageMagick/ImageMagick/assets/93042540/636892b2-b652-443c-b8ed-0b950f27917c)

**Comment by fmw42** [comment]
I do not understand what you want done. Please specify more clearly each part that you need removed.

**Reply by mediotex** [reply]
I mean to remove two handwritten marks at the top and seal stamp at the bottom. The coordinates seems are: 104x77+930+23, 141x187+542+14, 120x71+414+336

**Reply by fmw42** [reply]
Is this a one time activity or do you have other pages with similar markings to be removed. If the latter, are the coordinates the same as these?

**Comment by fmw42** [comment]
Your coordinates are way off for the document you provided.

Is this what you want?


```
color=$( convert document.png -crop 500x200+1500+300 +repage -scale 1x1! -format "%[pixel:u.p]" info: )
convert document.png \
-size 298x426 xc:"$color" -geometry +1066+25 -compose over -composite \
-size 246x178 xc:"$color" -geometry +1846+31 -compose over -composite \
-size 506x280 xc:"$color" -geometry +1639+1344 -compose over -composite \
document_result.png

```
![document_result](https://github.com/ImageMagick/ImageMagick/assets/13458218/c256d4d8-5773-4db7-b496-221951703668)

**Comment by mediotex** [comment]
These marking and stamp are on the first page only, the rest pages are without them.
I looked for object coordinates from IM GUI: _Image Edit_ > _Region of interest_, then I drew a rectangles around the target marks, and IM showed coordinates in small window.. 
But I mean to replace these marks not with a solid color, but with a scanned background copied from nearby areas of a similar color, say from an area I marked in green rectangle. There is one more place, that I want to hide use the same way: a glued paper bookmark: I drew a red ellipse around it.
![imageedit_6_7489010852](https://github.com/ImageMagick/ImageMagick/assets/93042540/125af9ff-a613-48fd-abdb-2d6d6f2b4226)

**Comment by fmw42** [comment, marked as answer]
Is this what you want?

```
convert document.png -crop 500x250+1400+250 +repage -duplicate 1 -append tile.png
convert document.png \
\( tile.png -crop 298x426+0+0 \) -geometry +1066+25 -compose over -composite \
\( tile.png -crop 246x178+0+0 \) -geometry +1846+31 -compose over -composite \
\( tile.png -crop 506x280+0+0 \) -geometry +1639+1344 -compose over -composite \
\( tile.png -crop 66x428+0+0 \) -geometry +1037+1200 -compose over -composite \
document_result2.png
rm -rf tile.png

```

![document_result2](https://github.com/ImageMagick/ImageMagick/assets/13458218/6543ec43-e8d3-4e51-8537-27f642f2dc11)

**Comment by mediotex** [comment]
Yes, that's what I meant.  Can you explain the code a little more? For example, the first two lines: in particular, what are the indicated values refer to. Thank you.
```
convert document.png -crop 500x250+1400+250 +repage -duplicate 1 -append tile.png
convert document.png \
\( tile.png -crop 298x426+0+0 \) -geometry +1066+25 -compose over -composite \
```

**Reply by fmw42** [reply]
500x250+1400+250 is the crop coordinates WxH+X+Y where I pulled out your background texture.  I duplicated it and added the duplicated to the crop in the vertical direction (-append) to have more room for your areas that you wanted to cover.

 298x426+0+0 is the size of the area you wanted to cover and will be cropped from the tiled background texture image.  +1066+25 is the offset of where to paste that in the large image.

Does that make sense to you now?

**Comment by mediotex** [comment]
Yes, thank you for the very helpful info.

**Reply by fmw42** [reply]
Try using GIMP to see where those coordinates are located.
